tests/Functional/BreadcrumbRenderTest.php

```diff
@@ -0,0 +1,113 @@
+      'oe_news_types' => 'http://publications.europa.eu/resource/authority/resource-type/ARTICLE_NEWS',
+      'oe_teaser' => 'News teaser',
+      'oe_summary' => 'http://www.example.org is a web page',
+      'body' => 'News body',
+      'oe_reference_code' => 'News reference',
+      'oe_publication_date' => [
+        'value' => '2020-09-18',
+      ],
+      'oe_author' => 'http://publications.europa.eu/resource/authority/corporate-body/ACJHR',
+      'oe_content_content_owner' => 'http://publications.europa.eu/resource/authority/corporate-body/COMMU',
+      'uid' => 0,
+      'status' => 1,
+    ]);
+    $node_1->save();
+
+    // Create a news article with different title.
+    $node_2 = $this->getStorage('node')->create([
+      'type' => 'oe_news',
+      'title' => 'Test news article breadcrumb',
+      'oe_news_types' => 'http://publications.europa.eu/resource/authority/resource-type/ARTICLE_NEWS',
+      'oe_teaser' => 'News teaser',
+      'oe_summary' => 'http://www.example.org is a web page',
+      'body' => 'News body',
+      'oe_reference_code' => 'News reference',
+      'oe_publication_date' => [
+        'value' => '2020-09-18',
+      ],
+      'oe_author' => 'http://publications.europa.eu/resource/authority/corporate-body/ACJHR',
+      'oe_content_content_owner' => 'http://publications.europa.eu/resource/authority/corporate-body/COMMU',
+      'uid' => 0,
+      'status' => 1,
+    ]);
+    $node_2->save();
+
+    $this->drupalGet($node_1->toUrl());
+    $this->assertBreadcrumbs('Test news node');
+
+    $this->drupalGet($node_2->toUrl());
+    $this->assertBreadcrumbs('Test news article breadcrumb');
+  }
+
+  /**
+   * Helper to assert both system and page header breadcrumbs on the page.
+   *
+   * @param string $last_segment_title
+   *   The expected last segment.
+   */
+  protected function assertBreadcrumbs(string $last_segment_title): void {
+    $breadcrumbs = [
+      // Test the page header breadcrumb.
+      '.ecl-page-header-core .ecl-breadcrumb-core',
+      // Test the system breadcrumb.
+      '[class="ecl-breadcrumb-core"]',
+    ];
+
+    foreach ($breadcrumbs as $breadcrumb) {
+      // Check for the number of segments.
+      $page_breadrumb = $this->assertSession()->elementExists('css', $breadcrumb);
+      $segments = $page_breadrumb->findAll('css', '.ecl-breadcrumb-core__segment');
+      $this->assertCount(3, $segments);
+
+      // Check for the number of links.
+      $links_count = $page_breadrumb->findAll('css', 'ol.ecl-breadcrumb-core__container li.ecl-breadcrumb-core__segment a.ecl-breadcrumb-core__link');
+      $this->assertCount(2, $links_count);
+
+      // Check the last segment.
+      $current_page = $page_breadrumb->findAll('css', 'ol.ecl-breadcrumb-core__container li.ecl-breadcrumb-core__current-page');
+      $this->assertCount(1, $current_page);
+      $current_page = reset($current_page);
+      $this->assertEquals($last_segment_title, trim($current_page->getText()));
+    }
```

Comment:
The rendering of the breadcrumb is tested in `\Drupal\Tests\oe_theme\Kernel\BreadcrumbTest`. Here we just need to assert the titles of the elements, so we can verify the caching.